src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (tail != null) {
+            tail.next = node;
+        }
+        tail = node;
+        if (head == null) {
+            head = node;
+        }
```

Comment:
this is quite complicated

basically you have 2 cases:

1. list is empty - then you init both head and tail like `head = tail = new Node()`
2. list is not empty - then you simply add an element at the end (consider creating a method like `addLast()`, you'll be able to  reuse it)